Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift

```diff
@@ -0,0 +1,487 @@
+      documentation: "Documentation for BService",
+      name: "BService",
+      namespace: "bnamespace",
+      methods: []
+    )
+
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for AService",
+      name: "AService",
+      namespace: "anamespace",
+      methods: []
+    )
+
+    let expectedSwift =
+      """
+      enum anamespace {
+          enum AService {
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = anamespace_AServiceServiceStreamingProtocol
+              typealias ServiceProtocol = anamespace_AServiceServiceProtocol
+          }
+      }
+      enum bnamespace {
+          enum BService {
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = bnamespace_BServiceServiceStreamingProtocol
+              typealias ServiceProtocol = bnamespace_BServiceServiceProtocol
+          }
+      }
+      """
+
+    try self.assertTypealiasTranslation(
+      codeGenerationRequest: self.makeCodeGenerationRequest(services: [serviceB, serviceA]),
+      expectedSwift: expectedSwift
+    )
+  }
+
+  func testTypealiasTranslatorNamespaceNoNamespaceOrder() throws {
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for AService",
+      name: "AService",
+      namespace: "anamespace",
+      methods: []
+    )
+    let serviceB = ServiceDescriptor(
+      documentation: "Documentation for BService",
+      name: "BService",
+      namespace: "",
+      methods: []
+    )
+    let expectedSwift =
+      """
+      enum BService {
+          static let methods: [MethodDescriptor] = []
+          typealias StreamingServiceProtocol = BServiceServiceStreamingProtocol
+          typealias ServiceProtocol = BServiceServiceProtocol
+      }
+      enum anamespace {
+          enum AService {
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = anamespace_AServiceServiceStreamingProtocol
+              typealias ServiceProtocol = anamespace_AServiceServiceProtocol
+          }
+      }
+      """
+
+    try self.assertTypealiasTranslation(
+      codeGenerationRequest: self.makeCodeGenerationRequest(services: [serviceA, serviceB]),
+      expectedSwift: expectedSwift
+    )
+  }
+
+  func testTypealiasTranslatorSameNameServicesNoNamespaceError() throws {
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for AService",
+      name: "AService",
+      namespace: "",
+      methods: []
+    )
+    let serviceB = ServiceDescriptor(
+      documentation: "Documentation for BService",
+      name: "AService",
+      namespace: "",
+      methods: []
+    )
+
+    let codeGenerationRequest = self.makeCodeGenerationRequest(services: [serviceA, serviceB])
+    let translator = TypealiasTranslator()
+    self.assertThrowsError(try translator.translate(from: codeGenerationRequest)) {
```

Comment:
Just copy this from the core-tests: https://github.com/grpc/grpc-swift/blob/79865359b894e8334e0d3b520294ceb2dfc7bcfd/Tests/GRPCCoreTests/Test%20Utilities/XCTest%2BUtilities.swift#L40-L51